rust/src/domains/ffi.rs

```diff
@@ -481,21 +481,67 @@ pub extern "C" fn opendp_domains__user_domain(
         Fallible::from(util::into_owned(c_res)?)?.downcast::<bool>()
     });
 
-    Ok(AnyDomain::new(UserDomain { element, member })).into()
+    Ok(AnyDomain::new(ExtrinsicDomain { element, member })).into()
 }
 
 #[bootstrap(
-    name = "_user_domain_descriptor",
+    name = "_extrinsic_domain_descriptor",
     returns(c_type = "FfiResult<ExtrinsicObject *>")
 )]
-/// Retrieve the descriptor value stored in a user domain.
+/// Retrieve the descriptor value stored in an extrinsic domain.
 ///
 /// # Arguments
-/// * `domain` - The UserDomain to extract the descriptor from
+/// * `domain` - The ExtrinsicDomain to extract the descriptor from
 #[no_mangle]
-pub extern "C" fn opendp_domains___user_domain_descriptor(
+pub extern "C" fn opendp_domains___extrinsic_domain_descriptor(
     domain: *mut AnyDomain,
 ) -> FfiResult<*mut ExtrinsicObject> {
-    let domain = try_!(try_as_ref!(domain).downcast_ref::<UserDomain>()).clone();
+    let domain = try_!(try_as_ref!(domain).downcast_ref::<ExtrinsicDomain>()).clone();
     FfiResult::Ok(util::into_raw(domain.element.value.clone()))
 }
+
+#[bootstrap(
+    name = "_extrinsic_domain",
```

Comment:
In the last PR I mentioned being concerned about this as a possible backdoor. Scenario: A is using opendp+pysyft to access a dataset controlled by B. As B is reviewing the code from A, `contrib` or `honest-but-curious` will set off an alarm. A just wants to get the work done, and isn't even malicious, and but uses `_extrinsic_something`. Underscores are private-ish, but wouldn't necessarily set off any alarms for B. Neither would `extrinsic`.

I don't have any great ideas. Two so-so ideas:
- These methods go in a private module and we make it harder to access them from `prelude`.
- Instead of being globals, the features are scoped, and they can be disabled in an appropriate `with` block.

Reply:
Moved into separate module.